rclpy/rclpy/impl/rcutils_logger.py

```diff
@@ -0,0 +1,302 @@
+# Copyright 2017 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from collections import OrderedDict
+from collections import namedtuple
+import importlib
+import inspect
+import os
+import time
+
+_rclpy_logging = importlib.import_module('._rclpy_logging', package='rclpy')
+
+# Known filenames from which logging methods can be called (will be ignored in `_find_caller`).
+_internal_callers = []
+# This will cause rclpy filenames to be registered in `_internal_callers` on first logging call.
+_populate_internal_callers = True
+
+
+def _find_caller(frame):
+    """Get the first calling frame that is outside of rclpy."""
+    if _populate_internal_callers:
+        global _internal_callers
+        global _populate_internal_callers
+        # Populate the list of internal filenames from which logging methods can be called.
+        # This has to be done from within a function to avoid cyclic module imports.
+        import rclpy.logging
+        # Extend the list to preserve any filenames that may have been added by third parties.
+        # Note: the call to `abspath` will also resolve mixed slashes that can result on Windows.
```

Comment:
Using `realpath` will also resolve symlinks and therefore ensure that they compare equal even if one is a symlink of the other.

Reply:
thanks! I changed it in c4ae7a7577f7b4c21d4914562f06f79d69406f44 but only for the `_find_caller` function. the file path that will displayed in the logging call is [still using abspath](https://github.com/ros2/rclpy/blob/c4ae7a7577f7b4c21d4914562f06f79d69406f44/rclpy/rclpy/impl/rcutils_logger.py#L63) because I thought that would be more what users are expecting (but maybe not)